Automattic/wp-calypso
# Display WooCommerce styles during JP Onboarding when coming from the setup wizard

This PR detects when we are in the Jetpack onboarding wizard, but have entered through the WooCommerce setup wizard. This is so we can do some co-branding, and keep the connection experience consistent throughout the store setup process. See p6riRB-4rG-p2 and https://github.com/woocommerce/woocommerce-admin/issues/2152.

This is step one of the process. It only adds the ability to target styles for this case, and implements the base background/button styles. It does not update the copy yet, or display the Woo logo, etc. It is also behind a feature flag so we can iterate on these differences without affecting the current flows.

@jameskoster I believe I could use some help here making sure I'm using the correct colors, hovers, accents, etc. I only had the background, text color, and button color to work with. See the mixin.

To Test:

* Run `npm run test-client` and make sure all tests pass.
* Disconnect Jetpack `Jetpack > Dashboard > Connections > Site connection > Manage site connection > Disconnect`.
* Click “Setup Jetpack”.
* On the connection/authorization screen ( https://wordpress.com/jetpack/connect/authorize?client_id= ….. ) copy and paste the whole URL.
* Replace `https://wordpress.com` with ` http://calypso.localhost:3000` and replace the `from` parameter in the URL (it may be something like `connection-banner` or `landing-page-bottom`) with  `from=woocommerce-setup-wizard` . See Pastebin `213d5`  or ping me if you have issues.
* Hit enter, and you should see Woo Purples
* Disconnect
* Logout
* Try the same same URL logged out
* Test the standard Jetpack connection process, without the WooCommerce query string appended, and verify you see the standard Jetpack Onboarding styles.

Screenshots:

<img width="717" alt="Screen Shot 2019-05-20 at 2 56 57 PM" src="https://user-images.githubusercontent.com/689165/58112448-8b733a80-7bc1-11e9-8894-d7f7c9d06a56.png">
<img width="789" alt="Screen Shot 2019-05-20 at 2 57 14 PM" src="https://user-images.githubusercontent.com/689165/58112449-8b733a80-7bc1-11e9-90ff-ec3a5ac9b5a0.png">
